Chymyst2, speculative tut, and macro& (scalatest run conflict) removal

phderome opened the pull request:
1) Chymyst2 rewording (including r/reply renaming in a late example, line 447 used both r and reply when only one should have been used)
2) getting rid of macro & with => run import as a substitute
3) slightly speculative intro of tut in plugins.sbt and build.sbt

phderome committed:
1) Chymyst2 rewording (including r/reply renaming in a late example)
2) getting rid of macro & with => run import as a substitute
3) slightly speculative intro of tut in plugins.sbt and build.sbt

winitzki commented on docs/chymyst02.md:
```diff
@@ -64,20 +64,20 @@ Once the reaction starts, it will receive the value `n = 123` from the input mol
 The syntax for the reply action makes it appear as if the molecule `f` carries _two_ values - its `Unit` value and a special `reply` function, and that the reaction body calls this `reply` function with an integer value.
 However, `f` is injected with the syntax `f()` -- just as any other molecule with `Unit` value.
-The `reply` function appears only in the pattern-matching expression for `f` inside a reaction.
+The `reply` function appears only in the pattern matching expression for `f` inside a reaction.
```
Are you sure this should not be hyphenated? I think it's easier to read this way. I see many uses of "pattern-matching expression" when I google for it (but people occasionally also say "pattern matching expression").

phderome replied:
It's only my nose where I prefer pattern matching, it feels more human language and less techie. I can put hyphen back

phderome replied:
reverted

winitzki commented on docs/chymyst02.md:
````diff
@@ -386,7 +386,7 @@ run { case result(n) + done(x) =>
 ```
 
-To make the chemistry clearer, we can rewrite this as three reactions using pattern-matching:
+To make the chemistry clearer, we can rewrite this as three reactions using pattern matching:
````
Here it should be left unhyphenated, I agree.

phderome replied:
I don't like too much that compile-time can be compile time and run-time runtime or run time, but... Anyhow, you call the shots on those. I am just laying out my proposal.

winitzki replied:
Well, at least this is not German, where you would just say things like "the Reactionspatternmatchingsexpression for Blockingmoleculessyntax is shown in Followingchapterstext" ;)

I am trying to see whether the reader could misunderstand the text. "Runtime" and "run time" are two different things to me: a "runtime" is an auxiliary executable or library that is used at "run time".

phderome replied:
unhyphenated

winitzki commented on docs/chymyst02.md:
```diff
@@ -64,20 +64,20 @@ Once the reaction starts, it will receive the value `n = 123` from the input mol
 Here `T` is the type of value that the molecule carries, and `R` (for blocking molecules) is the type of the reply value.
 
-The pattern-matching expression for a blocking molecule of type `B[T,R]` has the form `case ... + f(v, r) + ...` where `v` is of type `T` and `r` is of type `R => Boolean`.
+The pattern matching expression for a blocking molecule of type `B[T,R]` has the form `case ... + f(v, r) + ...` where `v` is of type `T` and `r` is of type `R => Boolean`.
```
Here again "pattern-matching" is used as an adjective, so I think it's easier to read. Otherwise "The pattern maching expression for a molecule" can be misread as "The pattern that matches an expression for a molecule".

phderome replied:
reverted

winitzki commented on macros/src/test/scala/code/winitzki/jc/MacrosSpec.scala:
```diff
@@ -1,6 +1,7 @@
 
 import JoinRun._
+import Macros.{run => &}
 import Macros._
```
I'm not really sure what else we need from `Macros` - maybe some types? What if we remove `import Macros._`?

phderome replied:
I can try to remove Macros._ from MacrosSpec and see what happens.

phderome replied:
this works: import Macros.{getName, rawTree, m,b, run => &}

winitzki commented:
No further comments.

phderome committed:
1) Chymyst2 rewording (including r/reply renaming in a late example)
2) getting rid of macro & with => run import as a substitute
3) slightly speculative intro of tut in plugins.sbt and build.sbt

Sergei's feedback leads to:
4) edits to revert pattern-matching used as adjective and simpler import for MacrosSpec.scala to resolve run conflict with Scalatest.

winitzki commented:
👍
